synchronizer/batches.go

```diff
@@ -110,20 +111,27 @@ func (bs *BatchSynchronizer) handleReorgs(ctx context.Context) {
 	for {
 		select {
 		case r := <-bs.reorgs:
+			bs.syncLock.Lock()
```

Comment:
Sorry for my ignorance, but isn't the Postgres DB already secured with the locks both during reading and writing?

Reply:
True, but that is not the point. The point being is that we can be processing some batches from the syncronizer, that start for example, from block 10, till block 15. In the meantime, the reorg detector, detects a reorg and updates the start block to 5. We will have inconsistent and possibly invalid batches in db because of that.